### PR TITLE
Add some new functions to the standard library

### PR DESCRIPTION
**Pull request by lorenzosinisi**
I have added the following functions:

 - MID
 - SWITCH
 - REMOVE_LAST_WORD
 - ROUND

The function MIX, SWITCH and ROUND are done mirroring Excel functionality, instead REMOVE_LAST_WORD is done to mirror the existing REMOVE_FIRST_WORD

This should help adoption and code repetition over multiple projects using this library

### CODE REVIEW

**Comment by smn** [comment]
@lorenzosinisi whoa! thanks for taking the time to contribute this 🫶 !

Out of curiosity, how are you using this library currently? Would love to learn how it's being used, and hopefully you're finding it useful?

**Comment by lorenzosinisi** [comment]
Hi @smn thanks, and I hope the functions can be merged in the standard lib! 

I am currently using it to allow non-devs to write simple code to transform data, and yes it is useful!

**Comment by lorenzosinisi** [comment]
Hi @smn is there any chance to get this merged?

File: lib/expression/callbacks/standard.ex
Comment (smn) [inline lib/expression/callbacks/standard.ex]: How is this different from supplying the format directly to `datevalue` here? That function accepts a string formatting argument already: https://github.com/turnhub/expression/blob/81ea56047cd9fdade0d5daee9c89bd0e450fc461/lib/expression/callbacks/standard.ex#L119-L149

Screenshot 2024-08-27 at 11 18 07

I can see the `text()` function name here being ambiguous and likely a source of confusion for folks given this seems like a generic function but is only really useful for date / datetime things.

That said, if you're wanting to use this in your own code, you could perhaps include it in a custom callbacks implementation ? The implementation isn't ideal but here's an example test case that does exactly that: https://github.com/turnhub/expression/blob/develop/test/expression_custom_callbacks_test.exs

File: lib/expression/callbacks/standard.ex
Comment (smn) [inline lib/expression/callbacks/standard.ex:923]: this one's great as it complements the already existing `remove_first_word()`

File: lib/expression/callbacks/standard.ex
Comment (smn) [inline lib/expression/callbacks/standard.ex:127]: thank you! This aligns with the excel function right? https://support.microsoft.com/en-us/office/switch-function-47ab33c0-28ce-4530-8a45-d532ec4aa25e

File: lib/expression/callbacks/standard.ex
Comment (smn) [inline lib/expression/callbacks/standard.ex:145]: same here, aligning with https://support.microsoft.com/en-us/office/round-function-c018c5d8-40fb-4053-90b1-b3e7f61a213c

File: lib/expression/callbacks/standard.ex
Comment (smn) [inline lib/expression/callbacks/standard.ex:183]: awesome, aligns with https://support.microsoft.com/en-us/office/mid-midb-functions-d5f9e25c-d7d6-472e-b568-4ecb12433028

File: lib/expression/callbacks/standard.ex
Comment (smn) [reply lib/expression/callbacks/standard.ex]: Reading https://support.microsoft.com/en-us/office/text-function-20d5ac4d-7b94-49fd-bb38-93d29371225c I understand why it's called `text()`, I guess it's a generic formatting function.

However, if we want to implement this we'll want to make sure to support other types of values as well. Until that is ready I would suggest keeping this one in a custom callbacks implementation?

**Review by smn — CHANGES_REQUESTED** [review]
Thanks so much for your contribution! It's highly appreciated. Before we merge this, can we discuss the implementation of `text()` ? 

If you're unable to complete it then we can merge the others in the mean time and you can keep the `text()` implementation in a custom callback module.

If you're able to complete it then that's awesome, we'd love to see a complete implementation of `text()` merged.

It's up to you!

File: lib/expression/callbacks/standard.ex
Comment (lorenzosinisi) [reply lib/expression/callbacks/standard.ex:127]: yes indeed, matches "SWITCH" 👍

File: lib/expression/callbacks/standard.ex
Comment (lorenzosinisi) [reply lib/expression/callbacks/standard.ex]: I would say we could remove this from this MR and leave it as custom callback, sounds good to me!

**Comment by lorenzosinisi** [comment]
To simplify I have opted for removing (for now) the TEXT function, how does it look now?

**Comment by smn** [comment]
Given the CLA that's in place, are you happy for us to merge these and for them to fall under the Turn.io copyright as per the CLA.md document in the repo? It looks like the CLA github action is broken (I'm trying to fix it) but I also don't want to hold up this merging if you're happy with it. 

Give me a few hours to see if I can get this CLA to run again. Something with its permissions seems to have changed.

**Comment by smn** [comment]
ok it looks fixed, could you merge in the latest changes in the `develop` branch in your PR? It should kick off the process.

**Comment by lorenzosinisi** [comment]
Sounds good to me yes! Thank you, and the same goes for the other PR have opened this morning 👌👌thank you!L

**Comment by lorenzosinisi** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by lorenzosinisi** [comment]
recheck

**Comment by lorenzosinisi** [comment]
Hi @smn any chance of merging this one?

**Review by smn — APPROVED** [review]

**Comment by smn** [comment]
thanks @lorenzosinisi ! Apologies for the delay in the review + merge, appreciate the contribution!

**Pull request merged**
